File: android/app/src/main/java/com/certisgroup/cathy/reactutils/CathyPackage.java
/*
 * Copyright (c) 2019 Certis CISCO Security Pte Ltd
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * Certis CISCO Security Pte Ltd. ("Confidential Information").
 * You shall not disclose such Confidential Information and shall use
 * it only in accordance with the terms of the license agreement you
 * entered into with Certis CISCO Security Pte Ltd.
 */
package com.certisgroup.cathy.reactutils;

import androidx.annotation.NonNull;

import com.certisgroup.cathy.reactutils.modules.AppVersionModule;
import com.certisgroup.cathy.reactutils.modules.BuildVariantModule;
import com.certisgroup.cathy.reactutils.modules.NavigationBarModule;
import com.certisgroup.cathy.reactutils.modules.SoftInputModeModule;
import com.certisgroup.cathy.reactutils.modules.crypto.AesModule;
import com.certisgroup.cathy.reactutils.modules.crypto.DigestModule;
import com.certisgroup.cathy.reactutils.modules.crypto.RsaModule;
import com.certisgroup.cathy.reactutils.views.gridlist.GridListManager;
import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.Arrays;
import java.util.List;

/**
 * Package to register custom native RN modules and views
 *
 * @author Lingqi
 */
public class CathyPackage implements ReactPackage {

    @NonNull
    @Override
    public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactContext) {
        return Arrays.asList(
            new GridListManager()
        );
    }

    @NonNull
    @Override
    public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactContext) {
        return Arrays.asList(
            new AppVersionModule(reactContext),
            new BuildVariantModule(reactContext),
            new NavigationBarModule(reactContext),
            new SoftInputModeModule(reactContext),
            new AesModule(reactContext),
            new DigestModule(reactContext),
            new RsaModule(reactContext)
        );
    }
}

File: ios/cathy/RCTUtils/CathyPackage.m
/*
 * Copyright (c) 2019 Certis CISCO Security Pte Ltd
 * All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * Certis CISCO Security Pte Ltd. ("Confidential Information").
 * You shall not disclose such Confidential Information and shall use
 * it only in accordance with the terms of the license agreement you
 * entered into with Certis CISCO Security Pte Ltd.
 */

#import "CathyPackage.h"
//#import "Cathy-Swift.h"

/* Package to register custom native RN modules and views */

@interface RCT_EXTERN_MODULE(AppVersion, NSObject)
@end

@interface RCT_EXTERN_MODULE(BuildVariant, NSObject)
@end

@interface RCT_EXTERN_MODULE(CryptoAES, NSObject)
RCT_EXTERN_METHOD(encrypt:(NSString *)mode :(NSString *)iv :(NSString *)key :(NSString *)data :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(decrypt:(NSString *)mode :(NSString *)iv :(NSString *)key :(NSString *)data :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
@end

@interface RCT_EXTERN_MODULE(CryptoDigest, NSObject)
RCT_EXTERN_METHOD(digest:(NSString *)algo :(NSString *)data :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
@end

@interface RCT_EXTERN_MODULE(CryptoRSA, NSObject)
RCT_EXTERN_METHOD(sign:(NSDictionary *)params :(NSString *)uuid :(NSString *)data :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(verify:(NSDictionary *)params :(NSString *)uuid :(NSString *)data :(NSString *)signature :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(encrypt:(NSDictionary *)params :(NSString *)uuid :(NSString *)data :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(decrypt:(NSDictionary *)params :(NSString *)uuid :(NSString *)data :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(generateKeyPair:(NSInteger)modulusSize :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(importKey:(NSString *)format :(NSString *)key :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(exportKey:(NSString *)format :(NSString *)uuid :(RCTPromiseResolveBlock)resolve :(RCTPromiseRejectBlock)reject)
@end

@interface RCT_EXTERN_MODULE(GridList, RCTViewManager)
RCT_EXTERN_VIEW_PROPERTY(itemCount, NSInteger)
RCT_EXTERN_VIEW_PROPERTY(rowHeight, CGFloat)
RCT_EXTERN_VIEW_PROPERTY(numColumns, NSInteger)
RCT_EXTERN_VIEW_PROPERTY(dragEnabled, BOOL)
RCT_EXTERN_VIEW_PROPERTY(verticalSpacing, CGFloat)
RCT_EXTERN_VIEW_PROPERTY(horizontalSpacing, CGFloat)
RCT_EXTERN_VIEW_PROPERTY(paddingInsets, NSDictionary *)
RCT_EXTERN_VIEW_PROPERTY(clipItem, BOOL)
RCT_EXTERN_METHOD(refreshDataSet:(NSNumber * _Nonnull)reactTag)
@end
